resources/User.py

```diff
@@ -1,69 +1,46 @@
-from werkzeug.security import generate_password_hash, check_password_hash
+from werkzeug.security import generate_password_hash
 from flask_restful import Resource
-from flask import request, jsonify
-import uuid
-import os
-import jwt
+from flask import request
+from middleware.user_queries import user_post_results
 
 
 class User(Resource):
     def __init__(self, **kwargs):
         self.psycopg2_connection = kwargs["psycopg2_connection"]
 
-    # Login function: allows a user to login using their email and password as credentials
-    # The password is compared to the hashed password stored in the users table
-    # Once the password is verified, an API key is generated, which is stored in the users table and sent to the verified user
-    def get(self):
+    def post(self):
+        """
+        Sign up function: allows a user to sign up by submitting an email and password.
+        The email and a hashed password are stored in the users table and this data is returned to the user upon completion
+        """
         try:
             data = request.get_json()
             email = data.get("email")
             password = data.get("password")
             cursor = self.psycopg2_connection.cursor()
-            cursor.execute(
-                f"select id, password_digest from users where email = '{email}'"
-            )
-            results = cursor.fetchall()
-            user_data = {}
-            if len(results) > 0:
-                user_data = {"id": results[0][0], "password_digest": results[0][1]}
-            else:
-                return {
-                    "message": "There username or password is incorrect. Please try again."
-                }, 400
+            user_post_results(cursor, email, password)
+            self.psycopg2_connection.commit()
 
-            if check_password_hash(user_data["password_digest"], password):
-                api_key = uuid.uuid4().hex
-                user_id = str(user_data["id"])
-                cursor.execute(
-                    "UPDATE users SET api_key = %s WHERE id = %s", (api_key, user_id)
-                )
-                payload = {
-                    "message": "API key successfully created",
-                    "api_key": api_key,
-                }
-                self.psycopg2_connection.commit()
-                return payload
+            return {"message": "Successfully added user"}
```

Comment:
One final thing here @mbodeantor, can we call the login endpoint from here to automatically log a user in once they've signed up? Or maybe as the API is concerned, all that means is to call the token middleware and return the token? I think it would be nice to automatically begin a session for new users rather than making them enter the same info to log in after signing up.

Reply:
Another option here is for the client to just automatically hit the login endpoint after signup. I think either way works, just lmk what you'd prefer.

Reply:
Yeah I think doing it automatically client side might make more sense. Not sure how common it is but like Max will be creating his own user from the API and won't need the login.

Reply:
Ah, right, that makes sense. And it keeps the API separated.